### PR TITLE
deuduplicated a few functions

### PR DESCRIPTION
**Pull request by rrzhang139**
parse_calldata and parse_code seem to have a few discrepancies between yaml and json version (e.g yaml does not mention asm as tags)

### CODE REVIEW

**Comment by aguzmant103** [comment]
Hey @rrzhang139  just checking, it this ready for review? Feel free to reach out directly or autoassign by assigning it to privacy-scaling-explorations/zkevm-reviewers

**Comment by rrzhang139** [comment]
@aguzmant103 I don't have the ability to assign reviewers, but this seems ready to review! Perhaps @ed255 could review this for me?

**Comment by ChihChengLiang** [comment]
Hi @rrzhang139, I can help review this PR. Can you resolve the conflict meanwhile?

**Comment by rrzhang139** [comment]
@ChihChengLiang I resolved this, but I am failing tests when running `make test`. I also cloned a fresh instance of the repo from `main` and it also failed the exact same tests, which is very strange. I pushed it, for now, to check if it passes CI, but I'm also debugging locally at the moment. I will keep you updated when I find the problem.

**Comment by ChihChengLiang** [comment]
@rrzhang139 What are the failing tests you see in `make test`? I did see some error logs from `make test`, but those should come from the negative test cases. The error logs look annoying but should be fine.

The CI reported some failing Clippy checks. You might need to run `make clippy` and fix some bugs.

File: testool/src/statetest/parse.rs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+pub fn parse_address(as_str: &str) -> Result<Address> {
+    if let Some(hex) = as_str.strip_prefix("0x") {
+        Ok(Address::from_slice(
+            &hex::decode(hex).context("parse_address")?,
+        ))
+    } else {
+        Ok(Address::from_slice(
+            &hex::decode(as_str).context("parse_address")?,
+        ))
+    }
```
Comment (ChihChengLiang) [inline testool/src/statetest/parse.rs:24]: Nitpick: I see many functions that use `strip_prefix` have duplicated lines. We could deduplicate the expression further. It's fine if we don't do it now.
```suggestion
    let hex = as_str.strip_prefix("0x").unwrap_or(as_str);
    Ok(Address::from_slice(
        &hex::decode(hex).context("parse_address")?,
    ))
```

Comment (ChihChengLiang) [inline testool/src/statetest/parse.rs:24]: Nitpick: I see many functions that use `strip_prefix` have duplicated lines. We could deduplicate the expression further. It's fine if we don't do it now.
```suggestion
    let hex = as_str.strip_prefix("0x").unwrap_or(as_str);
    Ok(Address::from_slice(
        &hex::decode(hex).context("parse_address")?,
    ))
```

**Review by ChihChengLiang — APPROVED** [review]
Great work! The code is less duplicated and looks nicer now!
We can merge once the CI issues are all fixed.
I added some low-hanging fruit refactor opportunities. They might be out of the scope of this PR so feel free to ignore them.

**Comment by ChihChengLiang** [comment]
@rrzhang139 one more clippy fix to go.

**Comment by ChihChengLiang** [comment]
Thank you @rrzhang139 for the contribution!

**Review by Lizy87 — APPROVED** [review]

**Review by Lizy87 — APPROVED** [review]

**Pull request merged**